src/NewTools-Debugger/StDebuggerActionModel.class.st

```diff
@@ -285,14 +285,19 @@ StDebuggerActionModel >> proceedDebugSession [
 
 { #category : #actions }
 StDebuggerActionModel >> recompileMethodTo: aString inContext: aContext notifying: aNotifyer [
+	| methodContext |
 
 	aContext ifNil: [ ^ self ].
 	self session
 		recompileMethodTo: aString
 		inContext: aContext
 		notifying: aNotifyer.
-	previousASTScope := (aContext compiledCode sourceNodeForPC:
-		                     aContext pc) scope
+		
+	methodContext := aContext.
+	[ methodContext outerContext ] whileNotNil: [
+		methodContext := methodContext outerContext ].
```

Comment:
> What I do not like is that we rely on the outerContext, this is no problem now, but if we change to clean blocks (or even just have the outerContext when we have a non-local return), this would stop working. (but it would be ok for the current Pharo11/Pharo12).
> 
> If we look at #home, does your loop not do the same? And always returns the home context?
> 
> If we want the home, we might be able to use "aContext activeHome" here... as we are interested in the home context on the stack? If true, this would avoid the need for an otherContext to to be there

I think `#activeHome` should work. And with the home, will it work with clean blocks?

Reply:
yes, without clean blocks active, #home would be ok